<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Drew Duckett (duckettdj@gmail.com), Tara Pelletier, &amp; Bryan Carstens" />

<meta name="date" content="2019-09-17" />

<title>Using P2C2M.SNAPP</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Using P2C2M.SNAPP</h1>
<h4 class="author">Drew Duckett (<a href="mailto:duckettdj@gmail.com">duckettdj@gmail.com</a>), Tara Pelletier, &amp; Bryan Carstens</h4>
<h4 class="date">2019-09-17</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Posterior Predictive Checks of Coalescent Models (P2C2M) is an R package that conducts posterior predictive checks to determine if your data fit the model assumptions of the coalescent analysis you are performing.</p>
<p>P2C2M.SNAPP applies this approach to SNAPP, a coalescent phylogeny estimation program that uses single nucleotide polymorphisms (SNPs). The package helps to determine if your data violate assumptions of the Multispecies Coalescent Model (MSCM).</p>
<p>The following summary statistics can currently be used to compare posterior and posterior predictive distributions: the Robinson Foulds tree distance (RF:Robinson and Foulds, 1981), the standard deviation of posterior likelihoods (MLSD), and the pairwise Fst outlier test (PFST). The PFST statistic was used as an attempt to identify which lineages were sharing genes when a model violation was detected, but simulation testing has shown that this statistic may perform poorly.</p>
<p>The necessary tutorial materials used below can be obtained from the P2C2M GitHub page at [<a href="http://github.com/P2C2M/P2C2M_SNAPP" class="uri">http://github.com/P2C2M/P2C2M_SNAPP</a>].</p>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>P2C2M.SNAPP can be installed in R from the provided tar.gz file:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;path/to/P2C2M.SNAPP_1.0.0.tar.gz&quot;</span>, <span class="dt">repos =</span> <span class="ot">NULL</span>, <span class="dt">type =</span> <span class="st">&quot;source&quot;</span>)</code></pre></div>
<p><strong>Dependencies:</strong></p>
<p><em>R</em>:<br />
R (&gt;= 3.5.0),<br />
ape (&gt;= 5.3),<br />
ggplot2 (&gt;= 3.2.0),<br />
graphics (&lt;= 3.5.0),<br />
grDevices (&gt;= 3.5.0),<br />
gsubfn (&gt;= 0.7),<br />
ggtree (&gt;= 1.14.6) - not on CRAN (see below),<br />
KRIS (&gt;= 1.1.1),<br />
stats (&gt;= 3.5.0),<br />
utils (&gt;= 3.5.0)</p>
<p>The easiest way to install ggtree is with code from the Bioconductor website:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
  <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)

BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;ggtree&quot;</span>)</code></pre></div>
<p><em>Additional</em>:<br />
fastsimcoal2 (Excoffier et al., 2013: available at [<a href="http://cmpg.unibe.ch/software/fastsimcoal2/" class="uri">http://cmpg.unibe.ch/software/fastsimcoal2/</a>]) See fastsimcoal2 website for if you encounter errors - they may be due to the version of gcc on your computer.</p>
<p><em>System</em>: Mac users need to have XQuartz installed for fastsimcoal2 to run: available at [<a href="https://www.xquartz.org" class="uri">https://www.xquartz.org</a>]</p>
<p><strong>The fastsimcoal2 executable must be located in the same directory as the input data files.</strong></p>
</div>
<div id="input-files" class="section level1">
<h1>Input Files</h1>
<p>In order to execute P2C2M, a user must provide a directory with four different files:</p>
<ol style="list-style-type: decimal">
<li><p>The SNAPP formatted .xml file used to run the SNAPP analysis<br />
<em>Note: You can not have any other .xml file in your working directory. After generating posterior predictive datasets, those .xml files should also be in your working directory</em></p></li>
<li><p>The file containing the posterior distribution of species trees (must be named <strong><em>snap.trees</em></strong>)</p></li>
<li><p>The log file from the SNAPP analysis (must be named <strong><em>snap.log</em></strong>)</p></li>
<li><p>A metadata text file containing on separate lines the number of SNPs used, an estimated mutation rate (subs/site/gen), the number of posterior trees saved to the .trees file and the number of samples per taxa (file must be named <strong><em>samples.txt</em></strong>).<br />
<em>Note: each taxon number should be on a separate line, with the order of taxa matching their order in the .xml file. The number of samples for each should follow the taxa numbers, separated by a tab. Please see the example files.</em></p></li>
</ol>
<p>Please make sure all four files are in you current working directory.</p>
<div id="important-snapp-information" class="section level2">
<h2>Important SNAPP Information</h2>
<p>P2C2M.SNAPP has been designed with BEAST 2.5.0. Therefore, it is unlikely that P2C2M.SNAPP will work with versions below 2.0.0 due to differences in the xml structure between BEAST versions.</p>
<p>When running SNAPP with your empirical data, there is one requirement for the MCMC parameters in order for P2C2M to work. The <em>Store Every</em> parameter of the MCMC chain must equal the <em>Log Every</em> parameter in the tracelog and the <em>Log Every</em> parameter in the treelog, i.e. if you are storing every 1000 trees in the chain, you should log every 1000 in the tracelog and treelog.</p>
<p>Finally, P2C2M.SNAPP will use the priors, model parameters, and MCMC parameters of your empirical SNAPP analysis when creating the posterior predictive xml files.</p>
</div>
</div>
<div id="running-p2c2m.snapp" class="section level1">
<h1>Running P2C2M.SNAPP</h1>
<div id="the-run.p2c2m.snapp-function" class="section level2">
<h2>The run.p2c2m.snapp function</h2>
<p>A single function is necessary to conduct posterior predictive checks. The function is run as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">run.p2c2m.snapp</span>(<span class="dt">path_to_xml =</span> <span class="kw">tempdir</span>(), <span class="dt">xml_file =</span> <span class="st">&quot;snapp.xml&quot;</span>, 
                <span class="dt">sum_stat =</span> <span class="kw">c</span>(<span class="st">&quot;RF&quot;</span>, <span class="st">&quot;MLSD&quot;</span>, <span class="st">&quot;PFST&quot;</span>), <span class="dt">num_sims =</span> <span class="dv">100</span>, <span class="dt">dist_reps =</span> <span class="dv">100</span>, 
                <span class="dt">sample_unif =</span> <span class="ot">TRUE</span>, <span class="dt">delete_sims =</span> <span class="ot">FALSE</span>, <span class="dt">save_graphs =</span> <span class="ot">TRUE</span>, 
                <span class="dt">save_output =</span> <span class="ot">TRUE</span>, <span class="dt">run_mode =</span> <span class="dv">1</span>)</code></pre></div>
<p>The p2c2m.snapp function contains the following arguments (default values are shown in the code above):</p>
<ol style="list-style-type: decimal">
<li><p>path_to_xml - the path to your SNAPP output files. Also where P2C2M output files will be written to.</p></li>
<li><p>xml_file - the name of your SNAPP formatted xml file</p></li>
<li><p>sum_stat - a vector of the summary statistics to use<br />
<em>Note: We recommend using all three summary statistics. However, if you do not have the computational resources to perform 100 SNAPP analyses, you can simply use the PFST statistic, which will bypass the need for SNAPP analyses of the posterior predictive datasets. Unfortunately, the PFST statistic performed poorly in simulation testing, so do this at your own risk.</em></p></li>
<li><p>num_sims - the number of posterior predictive datasets to simulate. Increasing this value may capture more stochasticity in the data. However, this is also the number of SNAPP analyses to perform when using the RF and MLSD statistics, so increasing it will be very computationally intensive.</p></li>
<li><p>dist_reps - the number of trees from the posterior predictive distribution to compare to the posterior tree they were simulated from (for the RF statistic)</p></li>
<li><p>sample_unif - if TRUE, sample posterior at regular interval. Otherwise, randomly sample.</p></li>
<li><p>delete_sims - if TRUE, simulation files are deleted after .xml files are created</p></li>
<li><p>save_graphs - if TRUE, histograms for each summary statistic are written to a file</p></li>
<li><p>save_output - if TRUE, results are written to a file</p></li>
<li><p>run_mode - specifies whether to simulate data and prepare xml files (1) OR calculate summary statistics and compare between empirical and posterior predictive datasets (2)</p></li>
</ol>
<p><em>Note: With the RF and MLSD summary statistics, run.p2c2m.snapp assumes that the user runs the simulated dataset xml files in SNAPP (probably on a cluster) between run_mode = 1 and run_mode = 2. Tree files and log files from the simulated dataset SNAPP runs must be placed in the same working directory as the empirical files.</em></p>
</div>
<div id="lets-try-an-example" class="section level2">
<h2>Let’s try an example</h2>
<p>This dataset was simulated with gene flow between two taxa. Therefore, P2C2M.SNAPP should flag it as violating the MSCM model implemented in SNAPP. It contains 6 taxa, with 2 samples per taxa, and 2000 SNPs.</p>
<p>We start with the .xml file (snapp.xml), the log file (snap.log), and the tree file (snap.trees) in our working directory. First, we need to create the samples.txt file (one has been included for this example - open it up to see how it is formatted). Now we are ready for run_mode 1:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">run.p2c2m.snapp</span>(<span class="dt">path_to_xml =</span> <span class="kw">tempdir</span>(), <span class="dt">xml_file =</span> <span class="st">&quot;snapp.xml&quot;</span>, 
                <span class="dt">sum_stat =</span> <span class="kw">c</span>(<span class="st">&quot;RF&quot;</span>, <span class="st">&quot;MLSD&quot;</span>, <span class="st">&quot;PFST&quot;</span>), <span class="dt">num_sims =</span> <span class="dv">10</span>, <span class="dt">dist_reps =</span> <span class="dv">100</span>, 
                <span class="dt">sample_unif =</span> <span class="ot">TRUE</span>, <span class="dt">delete_sims =</span> <span class="ot">FALSE</span>, <span class="dt">save_graphs =</span> <span class="ot">TRUE</span>, 
                <span class="dt">save_output =</span> <span class="ot">TRUE</span>, <span class="dt">run_mode =</span> <span class="dv">1</span>)</code></pre></div>
<p>This will produce a number of files. For each simulated dataset, it will produce a .par file to tell fastsimcoal2 how to simulate the data, arlequin output from fastsimcoal2, and a .xml file to be analyzed with SNAPP.</p>
<p>It also produces an intermediate.txt file, which contains important information for running run_mode 2: <strong>Do not change or remove this file</strong>. <em>Note: Here we are only running 10 simulations for the sake of time and memory.</em></p>
<p>Ordinarily, you would run these simulated datasets with SNAPP (probably on a cluster - a script has been provided to guide you in automating this procedure). Once the SNAPP analyses are completed, move the simulated .xml, .log, and .trees files into your working directory. For this tutorial, these files are provided.</p>
<p>Now we will run the second part of P2C2M.SNAPP, which compares the empirical and posterior predictive data. Make sure all .xml, .log, and .trees files, and intermediate.txt are in your working directory. Then run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">run.p2c2m.snapp</span>(<span class="dt">path_to_xml =</span> <span class="kw">tempdir</span>(), <span class="dt">xml_file =</span> <span class="st">&quot;snapp.xml&quot;</span>, 
                <span class="dt">sum_stat =</span> <span class="kw">c</span>(<span class="st">&quot;RF&quot;</span>, <span class="st">&quot;MLSD&quot;</span>, <span class="st">&quot;PFST&quot;</span>), <span class="dt">num_sims =</span> <span class="dv">10</span>, <span class="dt">dist_reps =</span> <span class="dv">100</span>, 
                <span class="dt">sample_unif =</span> <span class="ot">TRUE</span>, <span class="dt">delete_sims =</span> <span class="ot">FALSE</span>, <span class="dt">save_graphs =</span> <span class="ot">TRUE</span>, 
                <span class="dt">save_output =</span> <span class="ot">TRUE</span>, <span class="dt">run_mode =</span> <span class="dv">2</span>)</code></pre></div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>P2C2M.SNAPP produces several outputs. First, results will be printed to the R console. For the RF and MLSD statistics, a p-value is displayed. For MLSD, we consider values <strong>&lt;</strong>0.05 to indicate a model violation. For RF, we consider values <strong>&gt;</strong>0.05 to indicate a model violation (although unconventional, this is not a typo and is due to how the value is calculated).</p>
<p><em>Note: In simulation testing, we never observed a significant RF value. However, RF values greater than 0.00 were only observed with simulated datasets that included gene flow. <strong>Therefore, an RF value &gt;0.00 may indicate a model violation, but should not be used as the only evidence of a violation.</strong></em></p>
<p>The results of the PFST statistic will also be displayed. If PFST does not indicate a model violation, it will display <em>None</em>. If a violation is detected, it will display the numbers of the taxa that appear to be exchanging genes, e.g. <em>2v3</em> to indicate taxa numbers 2 and 3 appear to be exchanging genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">P2C2M Completed

               Stat   p     
p_rf           <span class="st">&quot;RF&quot;</span>   <span class="dv">0</span>     
p_mlsd         <span class="st">&quot;MLSD&quot;</span> <span class="dv">0</span>     
pair_fst_viols <span class="st">&quot;PFST&quot;</span> <span class="st">&quot;None&quot;</span>
Comparisons likely to violate model<span class="op">:</span>
<span class="st"> </span>None</code></pre></div>
<p>Here we can see that the MLSD statistic was significant even though the other two statistics weren’t, indicating that the dataset might violate the Multispecies Coalescent model implemented in SNAPP.</p>
<p>These results are also output in a text file (<em>p2c2m.out</em>) and a csv file (<em>P2C2Mout.csv</em>).</p>
<p>Finally, a number of histograms are generated as PDF files. For the RF and MLSD statistics, these show the empirical summary statistic value compared to the posterior predictive values. For the PFST statistic, the PDF shows differences between empirical and posterior predictive datasets for each pairwise Fst comparison.</p>
</div>
</div>
<div id="acknowledgments" class="section level1">
<h1>Acknowledgments</h1>
<p>We thank Remco Bouckaert for answering questions about the SNAPP program and the Ohio Supercomputer Center for early access to the Pitzer computing cluster. This work was supported by the National Science Foundation (DBI-1661029).</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
